test/tools/quic_request/BUILD

```diff
@@ -0,0 +1,21 @@
+load("@base_pip3//:requirements.bzl", "requirement")
+load(
+    "//bazel:envoy_build_system.bzl",
+    "envoy_package",
+    "envoy_py_test_binary",
+)
+
+licenses(["notice"])  # Apache 2
+
+envoy_package()
+
+envoy_py_test_binary(
+    name = "quic_request",
```

Comment:
Should we perhaps call this http3_request since it's doing an HTTP-level request, not a transport-level request?

Reply:
@RyanTheOptimist this is really useful

curl doesnt provide a client out of the box so testing stuff is _hard_

we hit similar issues trying to add a sandbox

cc @nguyenptk

Reply:
> Should we perhaps call this http3_request since it's doing an HTTP-level request, not a transport-level request?

I started out calling things `http3` and it wasted quite a bit of my time because [this bloody thing](https://pypi.org/project/http3/) kept coming up in searches and it's not at all obvious from the documentation that it is, in fact, nothing at all to do with `HTTP/3`. For that reason I prefer `quic`, but I can change it if you feel strongly.

Reply:
I would definitely prefer http3, or maybe h3, to quic because they're at different layers. It'd be like if we had tcp_request when we were actually doing http2. (Not a great analogy. But with various Foo-on-QUIC specs happening, I think it's worth making it clear when we're doing HTTP/3 and when we're doing QUIC since they're not quite the same thing)

> curl doesnt provide a client out of the box so testing stuff is hard

Oh, when we say "end to end" test here, we're not talking about the c++ integration tests? Can you say more about the kinds of tests we're talking about? (I'm not pushing back on this PR ... I just don't understand it yet :>)

Reply:
> Oh, when we say "end to end" test here, we're not talking about the c++ integration tests? Can you say more about the kinds of tests we're talking about? (I'm not pushing back on this PR ... I just don't understand it yet :>)

The specific test I'm trying to augment with this is https://github.com/envoyproxy/envoy/blob/main/test/integration/hotrestart_handoff_test.py - a test in which an actual envoy instance is started up, then replaced with a new instance, testing the full behavior of hot restart.

Right now it's testing three things with regular http requests:
1. a connection before draining continues to work while draining.
2. a connection during draining goes to the new instance.
3. a connection after draining is complete goes to the new instance.

The goal is to test the same things with http/3 (with, right now, the expectation that points 1 and 2 won't work correctly)

Reply:
for ref the sandbox we were working on is here https://github.com/envoyproxy/envoy/pull/23981 - we didnt get so far mostly due to the lack of http3 client to test with